### PR TITLE
Add "httpBasicAuth" option

### PR DESCRIPTION
**Pull request by mgielda**
As per discussion in #56 

Tested on Linux 64-bit (Ubuntu 15.10 to be precise).
Mattermost Desktop version 1.07, Mattermost server version 1.4 
I did not create any tests since there aren't any for the other options. I would appreciate an example :)

### CODE REVIEW

**Comment by yuya-oc** [comment]
Thanks. Please note that these codes will be changed in future release because this is an ad hoc feature as I said at #56.

File: src/main.js [hunk 1]
```diff
@@ -49,6 +49,23 @@ var mainWindow = null;
...
+  var readlineSync = require('readline-sync');
+
+  var username = readlineSync.question('Username: ');
+  var password = readlineSync.question('Password: ', {
+    hideEchoBack: true // The typed text on screen is hidden by `*` (default). 
+  });
+
+  app.on('login', function(event, webContents, request, authInfo, callback) {
+    event.preventDefault();
+    console.log("Replacing default auth behaviour.");
```
Comment (yuya-oc) [inline src/main.js]: Why don't you use the prompt in the 'login' event? It appears when it's needed.

In this commit, the app doesn't show anything when the app is executed by double-click. Users must know that command-line is necessary to use this feature.

Comment (yuya-oc) [inline src/main.js]: like:
```js
app.on('login', function(event, webContents, request, authInfo, callback){
  event.preventDefault();
  var username = readlineSync.question('Username: ');
  var password = readlineSync.question('Password: ', {
    hideEchoBack: true // The typed text on screen is hidden by `*` (default). 
  });
  callback(username, password);
});
```

Besides, in the case, the config might not be necessary.

**Comment by mgielda** [comment]
I will create a new pull request as your solution is obviously better. It did not work for me with previous, asynchronous solutions but obviously as it is now just adding those few lines works much better.

**Pull request closed without merging**